custom/enikshay/reports/worker_activity.py

```diff
@@ -0,0 +1,55 @@
+from corehq.apps.es import UserES
+from corehq.apps.es.users import mobile_users
+from corehq.apps.reports.analytics.esaccessors import get_submission_counts_by_user
+from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn, DTSortType
+from corehq.apps.reports.filters.dates import DatespanFilter
+from corehq.apps.reports.standard import CustomProjectReport
+from corehq.apps.reports.standard.monitoring import WorkerActivityReport
+from corehq.apps.reports.util import get_simplified_users, numcell
+from custom.enikshay.reports.filters import EnikshayLocationFilter
+
+from django.utils.translation import ugettext as _
+
+
+class EnikshayWorkerActivityReport(WorkerActivityReport, CustomProjectReport):
+    slug = 'enikshay_worker_activity_report'
+    num_avg_intervals = 4
+
+    @property
+    def fields(self):
+        return DatespanFilter, EnikshayLocationFilter
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(_("User")),
+            DataTablesColumn(_("# Forms Submitted"), sort_type=DTSortType.NUMERIC),
+            DataTablesColumn(_("Avg # Forms Submitted"), sort_type=DTSortType.NUMERIC),
+            DataTablesColumn(_("Last Form Submission"))
+        )
+
+    @property
+    def users(self):
+        location_ids = EnikshayLocationFilter.get_value(self.request, self.domain)
+        user_query = UserES().domain(self.domain).filter(mobile_users())
+        if location_ids:
+            user_query = user_query.location(location_ids)
+        return get_simplified_users(user_query)
+
+    @property
+    def rows(self):
+        rows = []
+        submissions_by_user = get_submission_counts_by_user(self.domain, self.datespan)
+        avg_submissions_by_user = get_submission_counts_by_user(self.domain, self.avg_datespan)
```

Comment:
Running these queries over the whole domain is going to be prohibitively expensive. You should add in the `user_ids` kwarg to limit it to only those users being displayed.